test/test_helper.rb

```diff
@@ -62,46 +62,43 @@ class ActiveSupport::TestCase
 end
 
 module SignInHelper
-  attr_accessor :credentials
+  attr_accessor :api_token
 
   def sign_in_as(user)
-    auth_token = create(:auth_token, user:)
-    self.credentials = { 'x-device-id': auth_token.device_id, 'x-secret': auth_token.secret }
+    self.api_token = create(:auth_token, user:).generate_token_for(:api)
   end
 
   def sign_out
-    self.credentials = {}
+    self.api_token = nil
   end
 
-  def merge_args(args)
-    args ||= {}
-    args[:headers] = args[:headers] || {}
-    creds = credentials || {}
-    { as: :json }.merge(args || {}).merge(headers: creds.merge(args[:headers]))
+  def merge_args(**args)
```

Comment:
I had the feeling we were doing a lot of unnecessary checks here, so I simplified this quite a bit